src/filter.rs

```diff
@@ -269,54 +269,87 @@ mod simd {
 /// this does not operate on pixels but on raw bytes of a scanline.
 ///
 /// Details on how each filter works can be found in the [PNG Book](http://www.libpng.org/pub/png/book/chapter09.html).
+///
+/// The default filter is `Adaptive`, which uses heuristics to select the best filter for every row.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum Filter {
+    NoFilter,
+    Sub,
+    Up,
+    Avg,
+    Paeth,
+    Adaptive,
+}
+
+impl Default for Filter {
+    fn default() -> Self {
+        Filter::Adaptive
+    }
+}
+
+impl From<RowFilter> for Filter {
+    fn from(value: RowFilter) -> Self {
+        match value {
+            RowFilter::NoFilter => Filter::NoFilter,
+            RowFilter::Sub => Filter::Sub,
+            RowFilter::Up => Filter::Up,
+            RowFilter::Avg => Filter::Avg,
+            RowFilter::Paeth => Filter::Paeth,
+        }
+    }
+}
+
+impl Filter {
+    pub(crate) fn from_simple(compression: Compression) -> Self {
+        match compression {
+            Compression::NoCompression => Filter::NoFilter, // with no DEFLATE filtering would only waste time
+            Compression::Fastest => Filter::None,
```

Comment:
`Filter::None` combines very poorly with `FdeflateUtraFast`, since fdeflate assumes a specific hard-coded symbol distribution. I'd expect a substantially better compression ratio with `Filter::Up` with minimal performance difference